### PR TITLE
Update migration guide

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md [hunk 1]
```diff
@@ -440,8 +440,7 @@ using (var ts = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
...
-
+-   **Plugins** - In the previous library, Microsoft.Azure.ServiceBus, users could [register plugins](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Microsoft.Azure.ServiceBus/src/QueueClient.cs#L527) that would alter an outgoing message before serialization, or alter an incoming message after being deserialized. These extension points allowed users of the Service Bus library to use common OSS extensions to enhance their applications without having to implement their own logic, and without having to wait for the SDK to explicitly support the needed feature. For instance, one use of the plugin functionality is to implement the [claim-check pattern](https://www.nuget.org/packages/ServiceBus.AttachmentPlugin/) to send and receive messages that exceed the Service Bus message size limits. To achieve similar functionality with `Azure.Messaging.ServiceBus`, you can extend the various types as demonstrated in the [extensibility sample](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/servicebus/Azure.Messaging.ServiceBus/samples/Sample09_Extensibility.md).
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md:443]: Is this really a gap, though?   I kind of think of it as just a different way to achieve the same functionality.

**Review by jsquire — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md:443]: Fair point. @ramya-rao-a what do you think?

Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md:443]: Personally, I think it is still a "gap" in that this won't work natively with the existing ecosystem of plugins.

**Pull request merged**